R/derive_var_trtemfl.R

```diff
@@ -0,0 +1,237 @@
+#' Derive Treatment-emergent Flag
+#'
+#' Derive treatment emergent analysis flag (e.g., `TRTEMFL`).
+#'
+#' @param dataset Input dataset
+#'
+#'   The variables specified by `start_date`, `end_date`, `trt_start_date`,
+#'   `trt_end_date`, `initial_intensity`, and `intensity` are expected.
+#'
+#' @param new_var New variable
+#'
+#' @param start_date Event start date
+#'
+#'   *Permitted Values:* A symbol referring to a date or datetime variable of
+#'   the input dataset
+#'
+#' @param end_date Event end date
+#'
+#'   *Permitted Values:* A symbol referring to a date or datetime variable of
+#'   the input dataset
+#'
+#' @param trt_start_date Treatment start date
+#'
+#'   *Permitted Values:* A symbol referring to a date or datetime variable of
+#'   the input dataset
+#'
+#' @param trt_end_date Treatment end date
+#'
+#'   *Permitted Values:* A symbol referring to a date or datetime variable of
+#'   the input dataset or `NULL`
+#'
+#' @param end_window
+#'
+#'    If the argument is specified, events starting more than the specified
+#'    number of days after end of treatment, are not flagged.
+#'
+#'    *Permitted Values:* A non-negative integer or `NULL`
+#'
+#' @param ignore_time_for_trt_end
+#'
+#'   If the argument is set to `TRUE`, the time part is ignored for checking if
+#'   the event occurred more than `end_window` days after end of treatment.
+#'
+#'   *Permitted Values:* `TRUE`, `FALSE`
+#'
+#' @param initial_intensity Initial severity/intensity or toxity
+#'
+#'   If the argument is specified, events which start before treatment start and
+#'   end after treatment start (or are ongoing) and worsened (i.e., the
+#'   intensity is greater than the initial intensity), are flagged.
+#'
+#'   The values of the specified variable must be comparable with the usual
+#'   comparison operators. I.e., if the intensity is greater than the initial
+#'   intensity `initial_intensity < intensity` must evaluate to `TRUE`.
+#'
+#'   *Permitted Values:* A symbol referring to a variable of the input dataset
+#'   or `NULL`
+#'
+#' @param intensity Severity/intensity or toxity
```

Comment:
I think we should add to this section something like: "This derivation assumes AE data collection method as single record per AE with “initial” and “most extreme” severity/intensity recorded separately." - as some data collection methods create a new record each time the AE intensity changes, which would have to be collapsed into one before using this function.

Reply:
Added